File: leetcode/algorithms/Valid Anagram/Valid Anagram.cpp
#include <string>
using namespace std;

class Solution {
public:
  bool isAnagram(string s, string t) {
    if (s.size() != t.size()) {
      return false;
    }
    int m[26] = {0};
    for (char c : s) {
      m[c - 'a']++;
    }
    for (char c : t) {
      if (m[c - 'a'] <= 0) {
        return false;
      }
      m[c - 'a']--;
    }
    for (int b : m) {
      if (b != 0) {
        return false;
      }
    }
    return true;
  }
};

File: leetcode/algorithms/Valid Anagram/Valid Anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        c1, c2 = {}, {}
        for c in s:
            c1[c] = c1.get(c, 0) + 1
        for c in t:
            c2[c] = c2.get(c, 0) + 1
        return c1 == c2
